apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -2515,4 +2516,329 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
+							},
+							{
+								label: 'Queued',
+								value: '$queued',
+							},
+							{
+								label: 'On_hold',
+								value: '$onhold',
+							},
+						],
+					},
+				},
+			],
+			{ readPreference: readSecondaryPreferred() },
+		);
+	}
+
+	getConversationsByDepartment(start: Date, end: Date): AggregationCursor<{ data: { label: string; value: number }[] }> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						t: 'l',
+						departmentId: {
+							$exists: true,
+						},
+						ts: {
+							$gte: start,
+							$lt: end,
+						},
+					},
+				},
+				{
+					$lookup: {
+						from: 'rocketchat_livechat_department',
+						localField: 'departmentId',
+						foreignField: '_id',
+						as: 'department',
+					},
+				},
+				{
+					$group: {
+						_id: {
+							$arrayElemAt: ['$department.name', 0],
+						},
+						Chats: {
+							$sum: 1,
+						},
+					},
+				},
+				{
+					$group: {
+						_id: null,
+						data: {
+							$push: {
+								label: '$_id',
+								value: '$Chats',
+							},
+						},
+					},
+				},
+				{
+					$project: {
+						_id: 0,
+					},
+				},
+			],
+			{ readPreference: readSecondaryPreferred() },
+		);
+	}
+
+	getConversationsByTags(start: Date, end: Date): AggregationCursor<{ data: { label: string; value: number }[] }> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						t: 'l',
+						ts: {
+							$gte: start,
+							$lt: end,
+						},
+					},
+				},
+				{
+					$group: {
+						_id: { $ifNull: ['$tags', 'Unspecified'] },
+						Chats: {
+							$sum: 1,
+						},
+					},
+				},
+				{
+					$unwind: '$_id',
+				},
+				{
+					$lookup: {
+						from: 'rocketchat_livechat_tag',
+						localField: '_id',
+						foreignField: '_id',
+						as: 'tag',
+					},
+				},
+				{
+					$project: {
+						count: '$Chats',
+						tag: { $first: '$tag' },
+					},
+				},
+				{
+					$addFields: {
+						tagName: { $ifNull: ['$tag.name', '$_id'] },
+					},
+				},
+				{
+					$group: {
+						_id: null,
+						data: {
+							$push: {
+								label: '$tagName',
+								value: '$count',
+							},
+						},
+					},
+				},
+				{
+					$project: {
+						_id: 0,
+					},
+				},
+			],
+			{ readPreference: readSecondaryPreferred() },
+		);
+	}
+
+	getConversationsByAgents(start: Date, end: Date): AggregationCursor<{ data: { label: string; value: number }[] }> {
+		return this.col.aggregate(
+			[
+				{
+					$match: {
+						t: 'l',
+						ts: {
+							$gte: start,
+							$lt: end,
+						},
+					},
+				},
+				{
+					$group: {
+						_id: {
+							$ifNull: ['$servedBy._id', 'Unassigned'],
+						},
+						total: { $sum: 1 },
+					},
+				},
+				{
+					$lookup: {
+						from: 'users',
+						localField: '_id',
+						foreignField: '_id',
+						as: 'agent',
+					},
+				},
+				{
+					$set: {
+						agent: { $first: '$agent' },
+					},
+				},
+				{
+					$addFields: {
+						agentName: {
+							$ifNull: ['$agent.name', '$_id'],
```

Comment:
What happens if there is an agent handling this conversation but their name is undefined (I believe this is possible)? Could this be a problem?